eccp

```diff
@@ -0,0 +1,129 @@
+#w = int(str(params['w']),16)
+#j = int(str(params['j']),16)
+#t = int(str(params['t']),16)
+#a = int(str(params['a']),16)
+#b = int(str(params['b']),16)
+#q = int(str(params['q']),16)
+#r = int(str(params['r']),16)
+
+# combinations
+c_t1 = ['s','w','j','t']
+c_t2 = ['s','w','a','b','t']
+llr_t = ['s','q']
+bls_t = ['s','b']
+
+# prime candidate
+n = config.get('Candidate','N')
+print n
+n = int(n.replace("$",""), 16)
+
+print "==============="
+print "Candidate: "
+print n
+print "==============="
+
+section = str(1)
+run=True
+while run: 
+    if section in sections:
+        par_list = []
+        options = config.options(section)
+        print "==============="
+        print section
+        print "==============="
+        for option in options:
+            print option + '=' + config.get(section,option)
+            value = config.get(section,option)
+            # if starts with minus
+            params[str(option)] = value.replace("$","")
+            par_list.append(option)
+        print par_list
+        print params
+# TODO: add some clean function to the end of the section, clean values in params
+        if set(c_t1) == set(par_list):
+            print "start ----> c_t1"
+            s = int(str(params['s']),16)
+            w = int(str(params['w']),16)
+            t = int(str(params['t']),16)
+            j = int(str(params['j']),16)
+            a = (3 * j * (oneseventwoeight - j)) % n
+            b = (2 * j * (oneseventwoeight - j)**2) % n    
+            n = curve_test(n, s, w, a, b, t)
+            print "n:"
+            print n
+            print "end ------> c_t1"
+            par_set()
+        if set(c_t2) == set(par_list):
+            print "start -----> c_t2"
+            s = int(str(params['s']),16)
+            w = int(str(params['w']),16)
+            a = int(str(params['a']),16)
+            b = int(str(params['b']),16)
+            t = int(str(params['t']),16)
+            n = curve_test(n, s, w, a, b, t)
+            print "n:"
+            print n
+            print "end ------> c_t2"
+            par_set()
+        if  set(llr_t) == set(par_list):
+            print "start -----> llr_t"
+            s = int(str(params['s']),16)
+            q = int(str(params['q']),16)
+            n = lucas_lehmer_riesel_test(n, s, q)
+            print "n:"
+            print n
+            print "end ------> llr_t"
+            par_set()
+        if set(bls_t) == set(par_list):
+            print "start ------ > bls_t"
+            s = int(str(params['s']),16)
+            b = int(str(params['b']),16)
+            n = brillhart_lehmer_selfridge_test(n, s, b)
+            print "n:"
+            print n
+            print "end ------ > bls_t"
+            par_set()
+        section = str(int(section) + 1)
+        run=True
+    else:
+        print "-> end of sections"
+        run=False
+        
+try:
+    if n % 2 == 0:
+        raise ValueError("not prime")
+    for i in xrange(3, int(n**0.5)+1, 2):
```

Comment:
xrange is not present in Python 3, please add an alias that will make it work on py3